provider/core/src/request.rs

```diff
@@ -134,6 +139,10 @@ impl Writeable for DataLocale {
             sink.write_str("-u-")?;
             self.keywords.write_to(sink)?;
         }
+        if let Some(aux) = self.aux.as_ref() {
+            sink.write_char('$')?;
```

Comment:
`$` has special meaning in string literals in languages we want to target, e.g. Dart. Is there a problem with `/`?

Reply:
I can probably make `/` work but it doesn't automatically work in the sorting function `strict_cmp` (the hot path for data lookup). It's easier to implement `$` because `'$' < '-'`. I could also use `#` which has the same property.

If we decide that `/` is what we really want, I'll make it work.

Reply:
If the restriction is separator < `-`, there are `!#$%&'*+,`. My ranking of these is 
1) `+` or `&` are great because the auxiliary key adds something to the data locale
1) `#` is a nice neutral separator
1) `!`, `*` could be used as separators, but they're usually numeric operator
1) `%` and `$` have the escaping issue
1) `'` and `,` are too typographic

Reply:
Other options above `-` would be `@`, which we already use in data keys, and `;`.

Reply:
Implemented `+` as decided in https://github.com/unicode-org/icu4x/issues/3632#issuecomment-1682787106